### PR TITLE
Fix Gcc8 warnings

### LINKED ISSUE #2153: compiler warnings on gcc8

### PR DESCRIPTION
**Pull request by marehr**
resolves #2153

### CODE REVIEW

**Comment by h-2** [comment]
still produces new errors/warnings...

**Comment by marehr** [comment]
Yeah, I know. I'm on it.

**Comment by marehr** [comment]
@h-2 build passes now

File: include/seqan/basic/alphabet_simple_type.h [hunk 1]
```diff
@@ -135,26 +135,19 @@ class SimpleType
...
-    SimpleType() : value(0)
-    {}
-
-   
-    SimpleType(SimpleType const & other)
-    {
-        assign(*this, other);
-    }
+    SimpleType() = default;
+    SimpleType(SimpleType const &) = default;
```
Comment (rrahn) [inline include/seqan/basic/alphabet_simple_type.h]: Please add a defaulted Destructor.

File: include/seqan/basic/pair_bit_compressed.h [hunk 2]
```diff
@@ -86,16 +86,22 @@ struct Pair<T1, T2, BitPacked<BITSIZE1, BITSIZE2> >
...
 
     inline Pair(T1 const & _i1, T2 const & _i2) : i1(_i1), i2(_i2) {}
 
     template <typename T1_, typename T2_, typename TSpec__>
     // TODO(holtgrew): explicit?
-    inline Pair(Pair<T1_, T2_, TSpec__> const &_p)
-            : i1(getValueI1(_p)), i2(getValueI2(_p)) {}
+    inline Pair(Pair<T1_, T2_, TSpec__> const &_p) :
+        i1(getValueI1(_p)), i2(getValueI2(_p))
+    {}
```
Comment (rrahn) [inline include/seqan/basic/pair_bit_compressed.h]: Please add default Destructor.

File: include/seqan/basic/pair_packed.h [hunk 3]
```diff
@@ -79,23 +79,28 @@ struct Pair<T1, T2, Pack>
     // Members
     // ------------------------------------------------------------------------
 
-    T1 i1;
-    T2 i2;
+    T1 i1{};
+    T2 i2{};
```
Comment (rrahn) [inline include/seqan/basic/pair_packed.h]: You do not need member initialisation if an explicit default ctor is required.

File: include/seqan/basic/pair_packed.h [hunk 4]
```diff
@@ -79,23 +79,28 @@ struct Pair<T1, T2, Pack>
...
 
     Pair(T1 const & _i1, T2 const & _i2) : i1(_i1), i2(_i2) {}
 
     template <typename T1_, typename T2_, typename TSpec__>
     // TODO(holtgrew): explicit?
-    Pair(Pair<T1_, T2_, TSpec__> const &_p)
-            : i1(getValueI1(_p)), i2(getValueI2(_p)) {}
+    Pair(Pair<T1_, T2_, TSpec__> const &_p) :
+        i1(getValueI1(_p)), i2(getValueI2(_p))
+    {}
```
Comment (rrahn) [inline include/seqan/basic/pair_packed.h]: Please add default Destructor

File: include/seqan/system/file_async.h [hunk 5]
```diff
@@ -813,20 +813,17 @@ class File<Async<TSpec> >:
 struct AiocbWrapper :
     public aiocb
 {
-    AiocbWrapper()
-    {}
+    AiocbWrapper() = default;
+    AiocbWrapper(AiocbWrapper &&) = default;
 
     AiocbWrapper(AiocbWrapper & other, Move) :
```
Comment (rrahn) [inline include/seqan/system/file_async.h]: Can you check if this move hack is still required? Now, that we actually have move constructors?
If you want to, you can also open a new ticket for this.

**Comment by marehr** [comment]
@rrahn  default constructors added!

**Pull request merged**